build_tools/python/e2e_test_framework/definitions/common_definitions.py

```diff
@@ -88,10 +89,19 @@ class InputDataFormat(Enum):
 class DeviceSpec(object):
   """Benchmark device specification."""
   id: str
+
   # Device vendor name. E.g., Pixel-6.
   vendor_name: str
+
+  # Host environment where the IREE runtime is running. For CPU device type,
+  # this is usually the same as the device that workloads are dispatched to.
+  # With a separate device, such as a GPU, however, the runtime and dispatched
+  # workloads will run on different platforms.
+  host_environment: HostEnvironment
```

Comment:
Should this be a `HostEnvironmentInfo` instead? Do we want to limit it to only pre-defined host environments? If `HostEnvironmentInfo` is supposed to be an implementation detail, maybe mark it private with a leading underscore? Same with DeviceArchitecture

Reply:
I intended to define all HostEnvironmentInfo/DeviceArchitecture as enums. I think it's better to have a list of all supported combinations in the same place, so the tools/workflows know what to support.

Reply:
Ok, I think you should probably mark it private then.

Would maybe be nice if we didn't have to have two separate classes for this, but Python enums are weird. I'm not really sure what `@dataclass` on an enum would do since they both use class properties. I just tested and it looks like basically Enum eats the class fields and then dataclass does nothing. I think one option you do have though that is maybe nicer (?) is making the enum subclass the dataclass, which would enforce that all values are of that type: https://docs.python.org/3/howto/enum.html#planet. Not sure how useful that actually is though.

```python
class HostEnvironment(HostEnvironmentInfo, Enum):
  """Predefined host environment."""

  LINUX_X86_64 = ("linux", "x86_64")
  ANDROID_ARM_64 = ("android", "arm_64")
```

Reply:
The inheritance SG and makes it easier to access the dataclass fields. Done.